src/ledger-iframe-bridge.test.ts

```diff
@@ -0,0 +1,482 @@
+import { hasProperty } from '@metamask/utils';
+
+import {
+  IFrameMessageAction,
+  LedgerIframeBridge,
+} from './ledger-iframe-bridge';
+import documentShim from '../test/document.shim';
+import windowShim from '../test/window.shim';
+
+global.document = documentShim;
+global.window = windowShim;
+
+// eslint-disable-next-line no-restricted-globals
+type HTMLIFrameElementShim = HTMLIFrameElement;
+// eslint-disable-next-line no-restricted-globals
+type WindowShim = Window;
+
+/**
+ * Checks if the iframe provided has a valid contentWindow
+ * and onload function.
+ *
+ * @param iframe - The iframe to check.
+ * @returns Returns true if the iframe is valid, false otherwise.
+ */
+function isIFrameValid(
+  iframe?: HTMLIFrameElementShim,
+): iframe is HTMLIFrameElementShim & { contentWindow: WindowShim } & {
+  onload: () => any;
+} {
+  return (
+    iframe !== undefined &&
+    hasProperty(iframe, 'contentWindow') &&
+    typeof iframe.onload === 'function' &&
+    hasProperty(iframe.contentWindow as WindowShim, 'postMessage')
+  );
+}
+
+/**
+ * Simulates the loading of an iframe by calling the onload function.
+ *
+ * @param iframe - The iframe to simulate the loading of.
+ * @returns Returns a promise that resolves when the onload function is called.
+ */
+async function simulateIFrameLoad(iframe?: HTMLIFrameElementShim) {
+  if (!isIFrameValid(iframe)) {
+    throw new Error('the iframe is not valid');
+  }
+  // we call manually the onload event to simulate the iframe loading
+  return await iframe.onload();
+}
+
+const LEDGER_IFRAME_ID = 'LEDGER-IFRAME';
+
+describe('LedgerIframeBridge', function () {
+  let bridge: LedgerIframeBridge;
+
+  /**
+   * Stubs the postMessage function of the keyring iframe.
+   *
+   * @param bridgeInstance - The bridge instance to stub.
+   * @param fn - The function to call when the postMessage function is called.
+   */
+  function stubKeyringIFramePostMessage(
+    bridgeInstance: LedgerIframeBridge,
+    fn: (message: any) => void,
+  ) {
+    if (!isIFrameValid(bridgeInstance.iframe)) {
+      throw new Error('the iframe is not valid');
+    }
+
+    jest
+      .spyOn(bridgeInstance.iframe.contentWindow, 'postMessage')
+      .mockImplementation(fn);
+  }
+
+  beforeEach(async function () {
+    bridge = new LedgerIframeBridge();
```

Comment:
Nit: Generally we prefer to use helper functions and construct things per-test. This ensures the tests are readable individually without consulting the beforeEach, and it removes barriers to testing scenarios that have a different setup.